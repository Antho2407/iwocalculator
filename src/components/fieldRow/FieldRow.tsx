import React from "react";
import { FieldRowRoot, Label, Info } from "./FieldRow.components";

export interface FieldRowProps {
  label: string;
  info: string;
}

const FieldRow: React.FC<FieldRowProps> = ({ label, info, children }) => {
  return (
    <FieldRowRoot>
      <Label>{label}</Label>
      {children}
      <Info>{info}</Info>
    </FieldRowRoot>
  );
};

export default FieldRow;
