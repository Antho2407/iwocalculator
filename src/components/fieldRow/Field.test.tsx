import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import FieldRow, { FieldRowProps } from "./FieldRow";

describe("<Field />", () => {
  let wrapper: ShallowWrapper<FieldRowProps>;

  describe("When the component is mounted", () => {
    it("should render successfully", () => {
      wrapper = shallow(<FieldRow label="test-label" info="test-info" />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
