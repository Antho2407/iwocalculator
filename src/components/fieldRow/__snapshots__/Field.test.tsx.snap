// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field /> When the component is mounted should render successfully 1`] = `ShallowWrapper {}`;
